File: exercises/Exercise023.java
import java.util.ArrayList;

public class Exercise023 {

    public static void main(String[] args) {
        int maxAbundant = 28123;
        ArrayList<Integer> abundants = new ArrayList<>();
        int[] numbers = new int[maxAbundant];

        for (int i = 1; i <= maxAbundant; i++)
        {
            numbers[i-1] = i;
            if (isAbundant(i)) abundants.add(i);
        }

        int pairSum = 0;
        for (int i : abundants)
        {
            for (int j : abundants)
            {
                pairSum = i + j;
                if (pairSum <= maxAbundant)
                    numbers[pairSum - 1] = 0;
            }
        }

        long sum = 0;
        for (int n : numbers) sum += n;
        System.out.println(sum);
    }

    public static boolean isAbundant(long n)
    {
        if (n < 1) throw new IllegalArgumentException("Input n must be >= 1");
        int sum = 0;
        for (int i = 1; i < n; i++)
        {
            if (n % i == 0) sum += i;
        }
        return sum > n;
    }

}

File: python/exercise023.py
def is_abundant(n):
    sum = 0
    for i in range(1, n):
        if n % i == 0:
            sum += i
    return sum > n


maxAbundant = 28123
abundants = []
numbers = []

for i in range(1, maxAbundant + 1):
    numbers.append(i)
    if is_abundant(i):
        abundants.append(i)

pairSum = 0
for i in abundants:
    for j in abundants:
        pairSum = i + j
        if pairSum <= maxAbundant:
            numbers[pairSum - 1] = 0

sum = 0
for n in numbers:
    sum += n
print(sum)